internal/docker/config.go

```diff
@@ -31,14 +30,14 @@ type Config struct {
 
 // NewConfig creates a new config to be used when creating
 // a docker client
-func NewConfig(endpoint string, timeout time.Duration, excludedImages []string, apiVersion string) (*Config, error) {
+func NewConfig(endpoint string, timeout time.Duration, excludedImages []string, apiVersion string) *Config {
```

Comment:
Maybe we should remove this function, embed `docker.Config` in the components configurations, and add a `Validate` method to `docker.Config`? (This would be called by https://github.com/open-telemetry/opentelemetry-collector/blob/fce1d4619dd65e1c53c0b83ccc154f5063ac8612/component/config.go#L35 )

This would allow us to reuse more code

Reply:
Makes sense to me, it would simply the code and avoid duplications. Changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34699/commits/133d4297b7f7a9df965bcd1db75f58a707990202